iabtcf-encoder/src/main/java/com/iabtcf/encoder/TCStringEncoder.java

```diff
@@ -265,8 +265,8 @@ private String encodeAllowedVendors() {
         private String encodeCoreString() {
             BitWriter bitWriter = new BitWriter();
             bitWriter.write(version, CORE_VERSION);
-            bitWriter.write(created, CORE_CREATED);
-            bitWriter.write(updated, CORE_LAST_UPDATED);
+            bitWriter.writeDays(created, CORE_CREATED);
```

Comment:
The BitWriter is capable of writing Instants and does not require a "writeDays" method. The class TCStringEncoder is responsible for the "business" logic and should ensure that "created" and "updated" are rounded to the nearest day and pass the resulting Instant to BitWriter#write(Instant).

This rounding method can be made a utility function and can be re-used in your test code which would also eliminate the TestUtils#toDeciDays method.